File: frontend/src/app/model/doctorOff.ts
import {Doctor} from './doctor';
import {OffType} from './offType';
import {OffStatus} from './offStatus';

export class DoctorOff {
  id: number;
  type: OffType;
  status: OffStatus;
  doctor: Doctor;

  constructor(type: OffType, status: OffStatus, doctor: Doctor, id?: number) {
    this.type = type;
    this.status = status;
    this.doctor = doctor;
    this.id = id;

  }

}

File: backend/src/main/java/project_backend/model/DoctorOff.java
package project_backend.model;

import javax.persistence.*;

@Entity
public class DoctorOff {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    private OffType type;

    @Enumerated(EnumType.STRING)
    private OffStatus status;

    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private Doctor doctor;

    public DoctorOff()
    {

    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public OffType getType() {
        return type;
    }

    public void setType(OffType type) {
        this.type = type;
    }

    public OffStatus getStatus() {
        return status;
    }

    public void setStatus(OffStatus status) {
        this.status = status;
    }

    public Doctor getDoctor() {
        return doctor;
    }

    public void setDoctor(Doctor doctor) {
        this.doctor = doctor;
    }
}
